skbio/alignment/_path.py

```diff
@@ -0,0 +1,308 @@
+        """Unpack states into an array of bits."""
+        # This should be faster than the generic solution I guess.
+        # TODO: Pending benchmark and optimization.
+        if not np.all(np.isin(self.states, [0, 1, 2])):
+            raise ValueError(
+                "For pairwise alignment, `states` must only contain "
+                "zeros, ones, or twos."
+            )
+        return np.stack([self.states & 1, self.states >> 1])
+
+    def to_cigar(self, seqs=None):
+        """Get a CIGAR string representing the pairwise alignment path.
+
+        seqs: If provided, will distinguish match (=) and mismatch (X). Otherwise,
+            will uniformly note them as (mis)match (M).
+        """
+        cigar = ""
+        lengths = self.lengths
+        gaps = np.squeeze(self.states)
+        codes = ["M", "I", "D"]
+        if seqs is not None:
+            query = seqs[0]
+            ref = seqs[1]
+            for qchar, rchar in zip(query, ref):
+                if qchar == rchar:
+                    cigar += "="
+                elif qchar == "-":
+                    cigar += "I"
+                elif rchar == "-":
+                    cigar += "D"
+                else:
+                    cigar += "X"
+            return self._run_length_encode(cigar)
+        else:
+            for i, length in enumerate(lengths):
+                cigar += str(length) + codes[gaps[i]]
+            return cigar
+
+    @classonlymethod
+    def from_cigar(cls, cigar):
+        """Create a pairwise alignment path from a CIGAR string."""
+        # need to have ability for strings without 1's
+        # Make sure cigar is not empty.
+        if not cigar:
+            raise ValueError("CIGAR string must not be empty.")
+
+        # Determine whether or not to fix the arrays.
+        fix_arrs = False
+        if "=" in cigar or "X" in cigar:
+            fix_arrs = True
+
+        cigar = cigar.replace("=", "M").replace("X", "M")
+        lengths = []
+        gaps = []
+        current_length = 0
+        mapping = {"M": 0, "I": 1, "D": 2}
+        no_ones = True
+        for char in cigar:
+            if char.isdigit():
+                no_ones = False
+                current_length = current_length * 10 + int(char)
+            elif char in mapping:
+                if no_ones:
+                    lengths.append(current_length + 1)
+                else:
+                    lengths.append(current_length)
+                gaps.append(mapping[char])
+                current_length = 0
+                no_ones = True
+            else:
+                raise ValueError("Invalid characters in CIGAR string.")
+        if fix_arrs:
+            lengths, gaps = cls._fix_arrays(
+                lengths=np.array(lengths), gaps=np.array(gaps)
+            )
+        return cls(np.asarray(lengths), np.asarray(gaps), [0, 0])
+
+    def _run_length_encode(self, input_string):
+        """Perform run length encoding on a string."""
```

Comment:
```suggestion
        r"""Perform run length encoding on a string."""
```
Last one, make sure all of the docstrings are formatted appropriately